### PR TITLE
Shows Toast when debit card number's count is not in range of 12-19

### LINKED ISSUE #857: Fix: Show error or toast when account number is not enough
**Summary:**
In SetupUpiPIN, the account number of debit card accepts any number of digits even 1 digit. We should give a toast when the user doesnt enter sufficient amount of digits which is usually 16 or even 15 in case of Amex.

### PR DESCRIPTION
**Pull request by shobhit24**
## Issue Fix
Fixes #857 

## Screenshots
Debit Card Number count is less than 12  : 

lessThan12

When Debit Card Number count is more than 19  :

morethan19
 

When Debit Card Number is in range i.e. 12-19  :

inRange

## Description
If the Debit Card Number entered is not sufficient i.e. not in the range 12 - 19, it will show a toast message i.e. "Invalid Debit Card".

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java [hunk 1]
```diff
@@ -95,7 +95,8 @@ public void verifyDebitCardSuccess(String otp) {
 
     @Override
     public void verifyDebitCardError(String message) {
-        showToast(message);
+
+        showToast(getString(R.string.debit_card_error_message));
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java:99]: revert this change, please. `verifyDebitCardError` should be dynamic as it was, otherwise getting that string parameter doesn't make any sense.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java [hunk 2]
```diff
@@ -27,8 +27,12 @@ public void attachView(BaseView baseView) {
     @Override
     public void verifyDebitCard(String s, String s1, String s2) {
         String otp = "0000";
-        mDebitCardView.verifyDebitCardSuccess(otp);
-//        String message = "error";
-//        mDebitCardView.verifyDebitCardError(message);
+        if((s.length() < 12) || (s.length() > 19)) {
+            String message = "";
+            mDebitCardView.verifyDebitCardError(message);
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: There is no need for this `message` variable, after reverting your changes in `verifyDebitCardError` method then you can simply do like : `mDebitCardView.verifyDebitCardError(getString(R.string.debit_card_error_message))`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java [hunk 3]
```diff
@@ -27,8 +27,12 @@ public void attachView(BaseView baseView) {
     @Override
     public void verifyDebitCard(String s, String s1, String s2) {
         String otp = "0000";
-        mDebitCardView.verifyDebitCardSuccess(otp);
-//        String message = "error";
-//        mDebitCardView.verifyDebitCardError(message);
+        if((s.length() < 12) || (s.length() > 19)) {
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:30]: Here you have a check style violation that is why your build is failing.  Enter a space between `if` and `(`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java (hunk 2 above)
Comment (shobhit24) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: I have tried it, but it is showing "Cannot resolve 'getString(int)'". 
ss

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java (hunk 3 above)
Comment (shobhit24) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:30]: Oh, okay
I'll correct it.
Thank You

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java (hunk 2 above)
Comment (jawidMuhammadi) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: Oh, my mistake.  Do like this: `getContext().getString(R...)`

Comment (shobhit24) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: showing same with getContext()

Comment (shobhit24) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: It worked, thank you
Any other change needed?

Comment (jawidMuhammadi) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:32]: After committing your changes don't forget to squash your commits into a single commit. Please read these guidelines too: [How to get started](https://gist.github.com/laxyapahuja/e30921a9478bdf3a9e166a023ce1882c) and [PR guidelines](https://github.com/openMF/mobile-wallet/blob/master/.github/CONTRIBUTING.md#pull-requests)

**Comment by shobhit24** [comment]
@jawidMuhammadi please guide me through this, what I'm doing wrong?

**Comment by jawidMuhammadi** [comment]
> @jawidMuhammadi please guide me through this, what I'm doing wrong?

Here is  your check style report, do the change accordingly: 
[ant:checkstyle] /home/travis/build/openMF/mobile-wallet/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java:97:54: '{' should have line break after.
[ant:checkstyle] /home/travis/build/openMF/mobile-wallet/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:34: Line is longer than 100 characters (found 107).
[ant:checkstyle] /home/travis/build/openMF/mobile-wallet/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java:35:9: '}' should be on the same line.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java [hunk 4]
```diff
@@ -95,7 +95,7 @@ public void verifyDebitCardSuccess(String otp) {
 
     @Override
     public void verifyDebitCardError(String message) {
-        showToast(message);
+        showToast(message);
```
Comment (rachittt) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java:98]:
```suggestion
        showToast(message);
```

Comment (rachittt) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.java:98]: Hi! please remove this inessential 'space' after the semicolon. it is not required and removing it would decrease the number of files changed.

File: mifospay/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -159,6 +159,7 @@
     <string name="disable_account">Disable account</string>
     <string name="log_out">Log out</string>
     <string name="debit_card_details">Debit Card Details</string>
+    <string name="debit_card_error_message">Invalid Debit Card</string>
```
Comment (luckyman20) [inline mifospay/src/main/res/values/strings.xml:162]: In my opinion, be more specific with the error.

Comment (shobhit24) [reply mifospay/src/main/res/values/strings.xml:162]: More specific?
like, Invalid Debit card number?

Comment (rachittt) [reply mifospay/src/main/res/values/strings.xml:162]: Yes

Comment (shobhit24) [reply mifospay/src/main/res/values/strings.xml]: Done
Any other modifications needed?
